torch/onnx/symbolic_opset9.py

```diff
@@ -2027,3 +2022,18 @@ def group_norm(g, input, num_groups, weight, bias, eps, cudnn_enabled):
     # Norm has shape [N, C, *] so we reshape weight and bias to [C, *]
     axes = [i for i in range(1, len(input_sizes) - 1)]
     return add(g, mul(g, norm, g.op("Unsqueeze", weight, axes_i=axes)), g.op("Unsqueeze", bias, axes_i=axes))
+
+
+@parse_args('v', 'v', 'i')
+def _weight_norm(g, weight_v, weight_g, dim):
+    rank = weight_v.type().dim()
+    if rank:
+        # W = g * ((v) / ||v||)
+        # Compute norm_except_dim for l2 norm
+        axes = list(range(rank))
+        axes.remove(dim)
```

Comment:
We should support dim=None as well (computes a norm over the entire weight tensor).
Could you verify if axes.remove(dim) fails when dim=None, and add a test case with dim=None as well?

Reply:
Thanks Lara for mentioning.
Looks like there is a conflicting behavior in torch.nn.utils.weight_norm regarding None dim.
Where dim could be negative for backwards axes, but when dim = None, it's overwitten to -1
https://github.com/pytorch/pytorch/blob/0c48092b2270d56cdab327bd1ff0ca89f5b7d569/torch/nn/utils/weight_norm.py#L10

Reply:
@BowenBao Could you please take a look as well? Thanks.